docs/presigned_urls.md

```diff
@@ -0,0 +1,354 @@
+    :type object_path: string
+    :param expiration: The number of seconds the presigned URL is valid for.
+    :type expiration: int
+    :return: The presigned URL.
+    :rtype: string or None
+    """
+
+    # Create low-level S3 service client
+    s3_client = boto3.client("s3")
+
+    try:
+        # URL generated by AWS user with access to private object (i.e. the owner)
+        url = s3_client.generate_presigned_url(
+            ClientMethod="get_object", # HTTP Method
+            Params={
+                "Bucket": bucket_name, # S3 bucket
+                "Key": object_path # relative object path
+            },
+            ExpiresIn=expiration # URL duration in seconds
+        )
+    except ClientError as err:
+        print(err)
+        logging.error(err)
+        return None
+
+    # Return the presigned URL as a string
+    return url
+```
+Further information on this *boto3* function can be found [here](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-presigned-urls.html).
+
+Non-programmatically, a presigned URL can be generated using the S3 console or AWS Explorer for Visual Studio – [comprehensive documentation on presigned URLs can be found here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html). Once a presigned URL is created for a private object, it can be used within Chromoscope configuration files to temporarily visualize these private files.
+
+&nbsp;
+## Configuration File Creation Using Presigned URLs
+
+### Presigned URLs within Chromoscope
+
+The ability to create presigned URLs not only allows for controlled, temporary visualization/sharing of private data, but also provides mode of sharing large amounts of data stored on the AWS cloud. Presigned URLs can be (1) used directly within a Chromoscope configuration file, linking to data for individual samples (simple single-sample example below): 
+```json
+[
+    {
+        "id": "EXAMPLE_ID",
+        "cancer": "breast",
+        "assembly": "hg19",
+        "sv": "https://EXAMPLE_BUCKET.s3.us-east-1.amazonaws.com/myobject.somatic.sv.bedpe?AWSAccessKeyId=AKIAEXAMPLEXXX&Expires=604800&Signature=alA%WOIDHCSha",
+        "cnv": "https://EXAMPLE_BUCKET.s3.us-east-1.amazonaws.com/myobject.somatic.cna.annotated.tsv?AWSAccessKeyId=AKIAEXAMPLEXXX&Expires=604800&Signature=jj321dtg2s%3ibOGf"
+    }
+]
+```
+
+and/or (2) as a temporary URL for the configuration file itself (if it is saved as a private object within an S3 bucket), used as the argument for [the `external` parameter of a Chromoscope URL](url-parameters.md):
+```
+// format
+https://chromoscope.bio/?external=[PRESIGNED_URL_FOR_YOUR_CONFIG_FILE]
+
+// example
+https://chromoscope.bio/?external=https://EXAMPLE_BUCKET.s3.us-east-1.amazonaws.com/myobject.configfile?AWSAccessKeyId=AKIAEXAMPLEXXX&Expires=604800&Signature=ibOGfAovnhASUASsdasjj321
+```
+
+&nbsp;
+### Configuration File Creation Scripts: Overview
```

Comment:
Maybe I missed it, but I wasn't sure if this generates a config file from already generated preassigned URLs of datasets or also generates preassigned URLs of datasets.

Reply:
generates presigned URLs for both the cohort data itself, and a presigned URL for the new configuration file it creates using those URLs

Reply:
^ this is in the overview section of cohort config creation. let me know if this is still unclear